### PR TITLE
New Tools package, carved off from util package

### PR DESCRIPTION
**Pull request by DanVanAtta**
Move the map making tools to their own package tree. The updated package structure is in the screen shot below. Most of the existing classes were plucked out of 'util' and moved to 'tools.image' or 'tools.map.making'. The XML creator was moved to 'tools.map.xml.creator'

package_restructure

I went through the package imports of the tools classes to try and pull stuff in that was self contained. We still have references to a few other classes, but they look to be shared. There is on Swing UI component moved over, which will be nice to have distinctly out of the main client source code.

* Notes on what has change: For the most part everything is a straight file move, except:
- (important) replaced invocation of tool classes by name (as a string value) with a compile time checked method call: `class.getName()`
- (minor) replaced a wildcard import with single imports
- (minor) fixed a typo

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/759)

### CODE REVIEW

**Comment by ron-murhammer** [comment]
@DanVanAtta Looks like this needs rebased.

**Comment by DanVanAtta** [comment]
@ron-murhammer  - pls have a look at the contents of this change. The merge conflicts with this one could be error prone to resolve, and @frigoref  will likely want to integrate with these changes ASAP.

**Comment by DanVanAtta** [comment]
@frigoref  - heads up on the xml creator being moved, have a look here when you get the chance

File: src/games/strategy/triplea/ui/MapData.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
 import games.strategy.triplea.ResourceLoader;
 import games.strategy.triplea.image.UnitImageFactory;
 import games.strategy.ui.Util;
-import games.strategy.util.PointFileReaderWriter;
+import tools.map.making.PointFileReaderWriter;
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/ui/MapData.java]: Thoughts on MapData importing from tools package? Any clean way to avoid that?

File: src/tools/map/making/PointFileReaderWriter.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package games.strategy.util;
+package tools.map.making;
```
Comment (ron-murhammer) [inline src/tools/map/making/PointFileReaderWriter.java]: Maybe better to leave this one in games/strategy for now since its used by GameData.

File: src/games/strategy/triplea/ui/MapData.java (hunk 1 above)
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/MapData.java]: It's a good catch, we should not have standard tripleA code calling into the tools packages. Some options:
- split PointFileReaderWriter up in to two,  core code vs non-core.
- move PointFileReaderWriter back ot the core code section, `games.strategy.util`

**Comment by ron-murhammer** [comment]
@DanVanAtta I like the changes and direction this is going. Just one class we might want to leave where it was on.

I'd like us to consider actually going further and before the next release maybe rename all the packages. I think 'games.strategy' is kind of dumb. The root package of the project should probably be 'triplea'. So you'd have something like:
triplea.ai
triplea.common
triplea.engine
triplea.rules
triplea.tools
triplea.ui

File: src/games/strategy/triplea/ui/MapData.java (hunk 1 above)
Comment (ron-murhammer) [inline src/games/strategy/triplea/ui/MapData.java]: I'd say for this PR just move it back to core. Can always do a separate PR to split it up.

Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/MapData.java]: Sounds good, stay tuned for the update.

**Comment by DanVanAtta** [comment]
Let's discuss packages further in: https://github.com/triplea-game/triplea/issues/763

About releasing:
I don't think we will necessarily freeze the code. At some time we will decide we are ready to do release testing. The latest build at that time becomes the release candidate.  Either:
- things look good and we flip the 'prerelease' checkbox to 'release'
- we add a bugfix to the latest master, and repeat the release testing on the latest build

We will have some control over how much new stuff enters the pipeline once we start doing release testing, but we should still continue incorporating new changes. That set of new changes should be pretty small, so we will test that along with everything else there is to test.. 

For this PR: I need to do some work on it still

**Comment by DanVanAtta** [comment]
updated, moved PointFileReaderWriter back to core packages.

**Comment by ron-murhammer** [comment]
@DanVanAtta Looks like this needs rebased.

**Comment by DanVanAtta** [comment]
updated

**Comment by DanVanAtta** [comment]
@ron-murhammer is this good to go? Seems like it's in the right direction. Let's tackle more package re-org in multiple future PRs.

**Comment by DanVanAtta** [comment]
Now has merge conflicts.. It's important we get the tools fixes in and the classes moved. From there it is much easier to do future work.

**Comment by DanVanAtta** [comment]
updated.
to be clear, for this PR I would like to keep the scope limited to what is here. Let's use this as a stepping stone for the next step. Before we do anything else we need to do some careful work to make sure any package moves will be safe. Note that the tools class and package names were hardcoded, and fixed in this PR so that we could move the tools safely.
